File: geeksforgeeks/sorting/sorting.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
struct point
{
    int x, y;
};

bool myComparator( point a, point b)
{
    return ( a.x < b.x);
}

int main()
{
    // arrays
    int arr[] = {10, 20, 5, 8, 2};
    int n  = sizeof(arr)/sizeof(int);
    sort(arr, arr+n);
    for( int a: arr )
    {
        cout << a << " ";
    }
    cout << endl;

    // sort(arr, arr+n, comparator function)
    //  for vector 
    vector<int> v = { 5,7, 2,2,1};
    sort(v.begin(), v.end());
    for( int x: v)
    {
        cout << x << " ";
    }
    cout << endl;
    
    // for struct how sorting works 
    point p[] = {
        { 3, 9}, { 2 ,8}, {1,8}
    };

    sort(p, p+(sizeof(p)/sizeof(point)), myComparator);

    for( auto x: p )
    {
        cout << x.x << " " << x.y << endl;
    }
}

File: geeksforgeeks/sorting/Sorting.java
import java.util.Arrays;
import java.util.Collections;
public class Sorting
{
    public static void main(String[] args) {
        int arr[] = { 5,2,7,1,8,9};
        Arrays.sort(arr);
        System.out.println("Array 1: " + Arrays.toString(arr));

        Integer[] newarr = Arrays.stream( arr ).boxed().toArray( Integer[]::new );
        Arrays.sort(newarr, Collections.reverseOrder());
        System.out.println("Array 2 " +  Arrays.toString(arr));

        int a[] = { 6 , 5, 1, 8, 2, 0};
        Arrays.sort(a, 2, 4);
        System.out.println("Array 3 : " + a.toString());
    }
}